posthog-android/src/main/java/com/posthog/android/replay/PostHogReplayIntegration.kt

```diff
@@ -460,20 +459,45 @@ public class PostHogReplayIntegration(
         status.lastSnapshot = wireframe
     }
 
+    /**
+     * Adapted from https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/view/View.java;l=11620;bpv=0;bpt=1
+     */
     private fun View.isVisible(): Boolean {
-        // TODO: also check for getGlobalVisibleRect intersects the display
-        val visible = isShown && width >= 0 && height >= 0 && this !is ViewStub
-
-        // Between API 16 and API 29, this method may incorrectly return false when magnification
-        // is enabled. On other versions, a node is considered visible even if it is not on
-        // the screen because magnification is active.
-        if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.Q) {
-            return visible
-        }
+        try {
+            if (isAttachedToWindow) {
```

Comment:
this is vendoring the most important part of `onInitializeAccessibilityNodeInfo`
but we don't have control over the internal impl nor do I know if it requires the main thread or not since I cannot reproduce the original issue, so I copied and adapted and adapted a try-catch just in case